app/models/middleware_datasource.rb

```diff
@@ -5,6 +5,7 @@ class MiddlewareDatasource < ApplicationRecord
   serialize :properties
 
   include LiveMetricsMixin
+  include NewWithTypeStiMixin
```

Comment:
Some style points, even though we don't enforce them anywhere, but I thought I'd mention anyway.

- `include`s should probably be the first thing in a class definition, before any calls to any DSL.  I see the LiveMetricsMixin was already in place, so for now it's fine.
- NewWithTypeStiMixin is typically the very first include regardless of alphabetical sorting, mainly because it messes around with .new which is pretty invasive.  Putting it first calls it out visually.